1_fetch/src/get_feature_vars.R

```diff
@@ -1,5 +1,52 @@
+finalize_vars_conus <- function(prep1_conus, prep2_conus, drop_attr) {
+  
+  #'@description combines outputs from prep1 and prep2 conus functions to provide
+  #'full set of feature variables for all comids in conus
+  #'
+  #'@param prep1_conus filepaths for all variables except land cover 
+  #'land cover variables (from prep1_vars_conus)
+  #'@param prep2_conus filepath for land cover variables (from prep2_vars_conus)
+  #'@param retain_attr character list of COMID and other attributes used for 
+  #'left joins 
+  #'
+  #'@return data frame including all conus-wide comids and their feature variables
+  
+  data <- read_csv(prep2_conus, show_col_types = FALSE)
+  for (i in prep1_conus) {
+    data_temp <- read_csv(i, show_col_types = FALSE) %>%
+      select(-all_of(drop_attr))
+    data <- left_join(data, data_temp, by = "COMID")
+  }
+  
+  return(data)
+}
+
+get_nhd_conus_gdb <- function(outdir, seven_zip) {
+  
+  #'@description uses nhdPlusTools package to download nhd geodatabase
+  #'
+  #'@param outdir filepath for final data downloads
+  #'@param seven_zip filepath (full, not relative) to 7zip executable
+  #'
+  #'@return unzipped nhd geodatabase
+  
+  download_nhdplusv2(
+    outdir,
+    url = paste0("https://edap-ow-data-commons.s3.amazonaws.com/NHDPlusV21/",
+                 "Data/NationalData/NHDPlusV21_NationalData_Seamless", 
+                 "_Geodatabase_Lower48_07.7z"),
```

Comment:
Is it possible to specify a specific version of this file to grab from S3? Do you know if this link will always grab the most recent?

Reply:
@ajsekell do you know the answer to Jared's question?